compiler/loco/include/loco/IR/DataTypeTraits.h

```diff
@@ -35,6 +35,12 @@ template <DataType DT> struct DataTypeImpl
 };
 
 // TODO Support other enum values
+template <> struct DataTypeImpl<DataType::S4>
+{
+  // Use C++ int8_t type for 4bit integer
+  using Type = int8_t;
```

Comment:
```suggestion
  // Use C++ uint8_t type for 4bit integer
  using Type = uint8_t;
```
To store 2 4bit values in 8bit unsigned?

Reply:
In memory will have 4bit per item in 8bit for simple manipulation.